src/NuGet.Clients/NuGet.CommandLine/NuGet.CommandLine.csproj

```diff
@@ -84,5 +73,28 @@
     </EmbeddedResource>
   </ItemGroup>
 
+  <Target Name="IlmergeNugetExe" AfterTargets="Build" Condition="'$(BuildingInsideVisualStudio)' != 'true'">
```

Comment:
i think it should be after Build. You still want the ilmerge'd nuget.exe even if someone didn't pack the project.